### PR TITLE
[for review]layers: use std mutex

### PR DESCRIPTION
**Pull request by jeremy-lunarg**
Mark and I would like to move to std::mutex. Is this okay? If so, I'll move all the layers to this method.

### CODE REVIEW

File: layers/image.cpp [hunk 1]
```diff
@@ -357,18 +349,18 @@ vkCreateImage(VkDevice device, const VkImageCreateInfo *pCreateInfo, const VkAll
         result = device_data->device_dispatch_table->CreateImage(device, pCreateInfo, pAllocator, pImage);
     }
     if (result == VK_SUCCESS) {
-        loader_platform_thread_lock_mutex(&globalLock);
+        std::lock_guard<std::mutex> lock(::mutex);
```
Comment (lentinem) [inline layers/image.cpp]: Why do you need to global namespace mutex? I'd also prefer using globalLock as the mutex as it's more descriptive although in the interest of variable naming consistency i would vote for using global_lock.

**Comment by lentinem** [comment]
As I've said before, I'm very in favor of doing this in general and would like to see this used everywhere.

File: layers/image.cpp (hunk 1 above)
Comment (jeremy-lunarg) [inline layers/image.cpp]: The std namespace is sucked in at the top. Actually, it had a redundant using-directive and I removed it, but I'd be in favor of removing both and going with explicit std scope resolution throughout.

**Comment by jeremy-lunarg** [comment]
Sounds good to me. I'm fine with global_lock. I'll work on updating all the
layers.

File: layers/image.cpp (hunk 1 above)
Comment (lentinem) [inline layers/image.cpp]: Ah ok. +1 to removing both and using explicit scope resolution.

**Comment by tobine** [comment]
I agree with previous comments:

1. Kill "using std"
2. rename "mutex"->"global_lock"

Looks good to push with those changes. Just assign back to me if you want me to merge it.

**Pull request closed without merging**